### PR TITLE
WIP: Use vectordrawables

### PR DESCRIPTION
**Pull request by ligi**
Replacing a lot of png's by VectorDrawables for the following reasons:
 - smaller
 - potentially sharper on some densities
 - ability to change color ( e.g. when using day/night mode from AppCompat )

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -24,6 +24,8 @@ android {
 
     lintOptions {
         warning 'MissingTranslation'
+        warning 'StringFormatInvalid'
```
Comment (johnjohndoe) [inline app/build.gradle]: Is this intentional in this commit?

File: build.gradle [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
+    ext {
+        support_version="24.2.0"
+    }
+
     repositories {
         jcenter()
         maven { url "https://jitpack.io" }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.5.0'
```
Comment (johnjohndoe) [inline build.gradle]: Should this be removals be part of this commit?

File: app/build.gradle (hunk 1 above)
Comment (ligi) [inline app/build.gradle]: Yes - seems to be a new lint rule that comes with the updated sdk

File: build.gradle (hunk 2 above)
Comment (ligi) [inline build.gradle]: yes - updating to gradle 2.1.2 so removing 1.5.0

File: app/build.gradle (hunk 1 above)
Comment (johnjohndoe) [inline app/build.gradle]: Ah, it is classified as an error by them and you want it to be a warning?

File: build.gradle (hunk 2 above)
Comment (johnjohndoe) [inline build.gradle]: I meant all removals.

File: app/build.gradle (hunk 1 above)
Comment (ligi) [inline app/build.gradle]: yea and was breaking the build this way

File: build.gradle (hunk 2 above)
Comment (johnjohndoe) [inline build.gradle]: I would commit the update of the individual dependencies in separate commits. The updates are kind of hidden here.

**Comment by ligi** [comment]
Interestingly it clean-builds fine on my dev-pc but not on travis - there it fails with a lint-error - can't yet make sense of this

**Comment by ligi** [comment]
@johnjohndoe can you check if the lint error happens on your PC? Can't really make sense of it - lint errors should not happen only on one PC and not the other and as I cannot see the reports ( one reason I do not use travis anymore ) - it would be good to know if it works on your PC

**Comment by ligi** [comment]
hm the build also works with snap-ci ..
https://snap-ci.com/ligi/criticalmaps-android/pull/1/logs/defaultPipeline/1/EditMe?back_to=build_history

**Comment by johnjohndoe** [comment]
> ... check if the lint error happens on your PC ...

The current branch **builds successfully** on my machine regardless whether I comment out the `warning 'StringFormatInvalid'` or not. The only thing I had to do was to change build environment to **Java 8**.

**Comment by ligi** [comment]
thanks - really strange that lint does not work the same ..
PS: travis also uses jdk8 via oraclejdk8

**Comment by johnjohndoe** [comment]
I used OpenJDK. Do you want me to try Oracle?

**Comment by ligi** [comment]
on my build-pc is also openjdk
in travis openjdk because this was used before - wanted to be minimal
invasive ..

**Comment by ligi** [comment]
travis has no openjdk8 :-(

$ jdk_switcher use openjdk8
Sorry, but JDK 'openjdk8' is not known.
The command "jdk_switcher use openjdk8" failed and exited with 1 during .
Your build has been stopped.


can we please perhaps just replace travis against snapci or something?

**Comment by stephanlindauer** [comment]
one way of building this on travis it could be in a docker container that has openjdk8 installed. if this sounds reasonable, let me know and i can take care of that.
i also wouldnt mind switching to another ci but i would then probably also want to move all the other repos, just to have everything in one place.

**Comment by stephanlindauer** [comment]
the overhead of going down the docker route would be downloading and extracting the base image (586.7 MB). the docker binaries already ship with the travis worker so we dont have to download and install them.

**Comment by stephanlindauer** [comment]
image

;)

**Comment by ligi** [comment]
I would be happy with both solutions - docker or snap-ci for only android

**Comment by stephanlindauer** [comment]
i tried the docker way. adds 8 minutes to the total duration of our builds. lets rather try snap-ci, shall we?

**Comment by cbalster** [comment]
way to go travis:
https://github.com/criticalmaps/criticalmaps-android/pull/175/commits/97ab1f1310ba657c64b554c66a9ac0965fd97e2c

**Comment by ligi** [comment]
just rebased - could be merged now

**Comment by cbalster** [comment]
Looks good, also like the updated icons.

**Comment by stephanlindauer** [comment]
:+1:

**Pull request merged**